### ISSUE TITLE
[BUG] PolyAnnotation not working with data units

### ISSUE THREAD

**Issue by graeme-hattan** [bokeh/bokeh#9393]
Bokeh version: 1.4
Python version: 3.7.5
Browser: Firefox / Chromium
OS: Linux

When I try to add a PolyAnnotation to my chart using (default) data units, it is not rendered. Also, the axes and the tools disappear. If I use screen units, everything works as expected.

Example code:

```
from bokeh.io import output_file, save
from bokeh.layouts import column
from bokeh.models import PolyAnnotation
from bokeh.plotting import figure

fig = figure()
fig.x([0, 50, 100], [0, 50, 100])
#poly = PolyAnnotation(xs=[0,100,100], ys=[0,100,0], xs_units='screen', ys_units='screen')
poly = PolyAnnotation(xs=[0,100,100], ys=[0,100,0])
fig.add_layout(poly)
output_file('poly.html')
save(fig)
```

Chromium javascript console:
```
Uncaught Error: not implemented
    at i.render (bokeh-1.4.0.min.js:383)
    at i._paint_levels (bokeh-1.4.0.min.js:533)
    at i.paint (bokeh-1.4.0.min.js:533)
    at i.after_layout (bokeh-1.4.0.min.js:533)
    at i.compute_layout (bokeh-1.4.0.min.js:497)
    at i.build (bokeh-1.4.0.min.js:497)
    at i.renderTo (bokeh-1.4.0.min.js:497)
    at v (bokeh-1.4.0.min.js:605)
    at Object.t.add_document_standalone (bokeh-1.4.0.min.js:605)
    at v (bokeh-1.4.0.min.js:240)
```
Screenshot:

![image](https://user-images.githubusercontent.com/1633586/68546861-36c3b600-03d3-11ea-9e95-09fcf0b86bc6.png)

**Comment by bryevdv** [comment]
Right, `PolyAnnotation` only currently accepts screen coordinates:

<img width="505" alt="Screen Shot 2019-11-10 at 9 46 28 AM" src="https://user-images.githubusercontent.com/1078448/68548209-15e16d80-039f-11ea-8475-c752985f2f87.png">

And the reason for that it because the only reason it was added to begin with was to support the `PolySelectTool` which needs to be able to draw in screen coordinates. I don't suppose it would be too difficult to add support for data coordinates.

**Comment by graeme-hattan** [comment]
Thanks for the quick response! I was actually trying to use this as a (non ideal) way to get a filled slope as per feature request #9194.

I knew I was off the beaten track when there was no documented example. I got really confused not having it work with the default values, so even just having this documented and/or default to screen units might save somebody else some pain.

**Comment by bryevdv** [comment]
Sorry about the confusion, we can hopefully get one or both of these issues resolved fairly soon.

**Comment by mattpap** [comment]
This was fixed in PR #10833 in bokeh 2.3.
